src/common/abstractPromote.ts

```diff
@@ -101,6 +102,38 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     };
   }
 
+  protected async requestPromotionFlow(doceOrg: Org, numRetries: number): Promise<string> {
```

Comment:
@molson-sfdc I'll be working in this PR.
I find these two methods to be somewhat repetitive. Here is my suggestion to have the retry logic in one method.
```suggestion
protected async requestPromotionWithRetry(doceOrg: Org, numRetries: number): Promise<string> {
    try {
      return await this.requestPromotion(doceOrg);
    } catch (error) {
      if (error instanceof Error && error['errorCode'] === HTTP_CONFLICT_CODE && numRetries >= 1 ) {
        this.spinner.start('Sync of VCS Events In-Progress');
        return await this.requestPromotionWithRetry(doceOrg, numRetries - 1);
      }
      throw error;
    } finally {
      // we need to stop the spinner in case it was started in a retry attempt.
      this.spinner.stop();
    }
  }
```

Reply:
Won't this repeatedly call `startSpinner` ?

Reply:
Yes, it will.
We could use `this.spinner.status` to check if it is already spinning and start it if it's not.